plugins/catalog-backend-module-incremental-ingestion/src/database/IncrementalIngestionDatabaseManager.ts

```diff
@@ -546,12 +560,26 @@ export class IncrementalIngestionDatabaseManager {
    * @param entities - DeferredEntity[]
    */
   async createMarkEntities(markId: string, entities: DeferredEntity[]) {
+    const refs = entities.map(e => stringifyEntityRef(e.entity));
+
     await this.client.transaction(async tx => {
+      const existingRefs = (
+        await tx<{ ref: string }>('ingestion_mark_entities')
+          .select('ref')
+          .whereIn('ref', refs)
+      ).map(e => e.ref);
+
+      const newRefs = refs.filter(e => !existingRefs.includes(e));
+
+      await tx('ingestion_mark_entities')
+        .update('ingestion_mark_id', markId)
+        .whereIn('ref', existingRefs);
```

Comment:
Might be worth looking into efficient upserts in a future PR (not now). If the table has a UNIQUE constraint on ingestion_mark_id and ref as a pair, you could make an efficient upsert with just blindly inserting things and doing `.onConflict(['ingestion_mark_id', 'ref']).ignore()` or so. The same technique would be used in the global table I've been proposing, to track everything we're adding through the connection